sympy/functions/special/delta_functions.py

```diff
@@ -31,6 +49,38 @@ class DiracDelta(Function):
 
     5) ``DiracDelta(x,k) = 0``, for all ``x != 0``
 
+    Examples
+    ========
+
+    >>> from sympy import DiracDelta, diff, pi, Piecewise
+    >>> from sympy.abc import x, y
+
+    >>> DiracDelta(x)
+    DiracDelta(x)
+    >>> DiracDelta(1)
+    0
+    >>> DiracDelta(-1)
+    0
+    >>> DiracDelta(pi)
+    0
+    >>> DiracDelta(x - 4).subs(x, 4)
+    DiracDelta(0)
+    >>> diff(DiracDelta(x))
+    DiracDelta(x, 1)
+    >>> DiracDelta(x**2 - 1).fdiff()
+    DiracDelta(x**2 - 1, 1)
+    >>> diff(DiracDelta(x - 1),x,2)
+    DiracDelta(x - 1, 2)
```

Comment:
I think this would depict that we can differentiate as many times in a single expression.